docs/blog/20230501-Daeraxa-MayUpdate.md

```diff
@@ -0,0 +1,161 @@
+
+[@savetheclocktower] has been working on some improvements for this. For example, Tree-sitter parsers are quite good at this sort of code analysis, and can identify the important parts of a source code file with [a simple query file](https://tree-sitter.github.io/tree-sitter/code-navigation-systems).
+
+[Language servers](https://microsoft.github.io/language-server-protocol/) are another potential source of symbol information. Some language servers are even capable of supplying project-wide symbol information; this would improve other `symbol-view` responsibilities, like the “Go to Declaration” command.
+
+So to make steps towards achieving this, we'd refactor `symbols-view` to be just a generic UI for symbol navigation. We'd then package different “providers” of symbols as services for `symbols-view` to consume. This is very similar to how the `autocomplete-plus` package works.
+
+This has the potential to really upgrade this functionality and make navigating files and projects far more intuitive. Once the work for the `tree-sitter` upgrades have been completed then we will hopefully have something more concrete to share on this feature.
+
+## CI Testing speed upgrade
+
+<img src="./assets/ghtests.png"/>
+
+[@DeedeeG] has [made an update](https://github.com/pulsar-edit/pulsar/pull/492) to our continuous integration configuration by caching and restoring the dependencies which ends up saving us a huge amount of time, possibly over an hour each run! (Pulsar has to build and test an awful lot of packages on each run so it is a long process). This should hopefully make life just that bit easier on the development side of things.
+
+## Backend webhook
+
+<img src="./assets/package-webhook.png"/>
+
+[@confused-techie] has been busy adding yet another [new feature](https://github.com/pulsar-edit/package-backend/pull/156) to our package backend, this time it is a new webhook that allows us to publish messages and notifications when package authors publish and update their packages. You can check this out in action in our [backend-notifications](https://discordapp.com/channels/992103415163396136/1045131651748999239) channel on Discord.
+
+## Package website readme links overhaul
+
+<img src="./assets/package-frontend.png"/>
+
+Now some [changes to our package website](https://github.com/pulsar-edit/package-frontend/pull/96). There are a few separate, but related, changes by [@confused-techie] here that involve improving our package website to overhaul some of the issues we have with readme links.
+
+The first change adds shorthand string author assignment in the `package.json` file of a package. Previously if this was not populated fully then the author wouldn't display. This fixes that problem by allowing the partial and shorter form details.
+**Note**: do we have an example of what this shorter form looks like vs what was required?
```

Comment:
So essentially according to the [CommonJS Package Spec](https://wiki.commonjs.org/wiki/Packages/1.0) to define an author you would need an object containing: `name`, `email`, and `web`. 

The [NPM Specs](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#people-fields-author-contributors) expand on this, allowing what I've, lovingly, decided to call author shorthand, where you can use a string such as `author <email> (web)` instead of an object.

But when originally creating the parsing for this author object shorthand, I had never considered that the spec allows for the exclusion of both or some of the fields. With `author` being the only thing required within the author object shorthand.

So the change here, is what I decided to call partial shorthand, where only some of those fields are defined, which is what that PR did, brought support for partial shorthand author strings, (although in my implementation everything is optional, including the author. Unlike NPM's spec)

Reply:
Could you have a look again to see if this is accurate?

Reply:
Addressed a syntax error an an additional review, but otherwise, yes this looks perfectly accurate now!